rest-service/manager_rest/test/endpoints/test_deployment_labels_dependencies.py

```diff
@@ -1,613 +1,420 @@
+            labels=[{'csys-obj-type': 'service'}]
         )
-        self._create_deployment_objects('env', 'service', 2)
-        self._create_deployment_objects('env', 'environment', 2)
-
-        deployment = self.client.deployments.get('env')
-        self.assertEqual(deployment.sub_services_count, 2)
-        self.assertEqual(deployment.sub_environments_count, 2)
+        assert env1.sub_services_count == 0
+        assert env1.sub_environments_count == 1
 
+    def test_detach_all_environments_from_deployment(self):
+        env1 = self._deployment(id='env1')
+        self._deployment(id='env2')
+        self._deployment(id='srv1')
         self.client.deployments.update_labels(
-            'environment_1_env',
-            [
-                {
-                    'csys-obj-type': 'environment'
-                }
+            'env2',
+            labels=[
+                {'csys-obj-type': 'environment'},
+                {'csys-obj-parent': 'env1'}
             ]
         )
         self.client.deployments.update_labels(
-            'environment_2_env',
-            [
-                {
-                    'csys-obj-type': 'environment'
-                }
+            'srv1',
+            labels=[
+                {'csys-obj-parent': 'env1'}
             ]
         )
+        assert env1.sub_services_count == 1
+        assert env1.sub_environments_count == 1
 
-        deployment = self.client.deployments.get('env')
-        self.assertEqual(deployment.sub_services_count, 2)
-        self.assertEqual(deployment.sub_environments_count, 0)
+        self.client.deployments.update_labels(
+            'env2',
+            labels=[{'csys-obj-type': 'service'}]
+        )
+        assert env1.sub_services_count == 1
+        assert env1.sub_environments_count == 0
 
     def test_deployment_statuses_after_creation_without_sub_deployments(self):
-        self.put_deployment('dep1')
-        deployment = self.client.deployments.get('dep1')
-        self.assertEqual(
-            deployment.deployment_status,
-            DeploymentState.REQUIRE_ATTENTION
+        bp = models.Blueprint(
+            id='bp1', creator=self.user, tenant=self.tenant, plan={})
+        db.session.add(bp)
+        self.client.deployments.create('bp1', 'dep1')
+        dep1 = db.session.query(models.Deployment).filter_by(id='dep1').one()
+
+        assert dep1.deployment_status == DeploymentState.REQUIRE_ATTENTION
+        assert dep1.sub_services_status is None
+        assert dep1.sub_environments_status is None
+
+        exc = models.Execution(
+            id='exc1',
+            deployment=dep1,
+            workflow_id='create_deployment_environment',
         )
-        self.assertIsNone(deployment.sub_services_status)
-        self.assertIsNone(deployment.sub_environments_status)
+        db.session.add(exc)
+        self.client.executions.update(exc.id, status='terminated')
+        assert dep1.deployment_status == DeploymentState.GOOD
```

Comment:
adding this check too, for no good reason, but might as well - check that it's require_attn up front, and `good` after running create-dep-env.